src/syntax/syntax.rkt

```diff
@@ -400,18 +402,56 @@
 (define (clear-active-operator-impls!)
   (set-clear! active-operator-impls))
 
-;; Registers an operator implementation `name` or real operator `op`.
-;; The input and output representations must satisfy the types
-;; specified by the `itype` and `otype` fields for `op`.
-(define (register-operator-impl! op name ireprs orepr attrib-dict)
+;; Registers an operator implementation `name`
+;; fl, spec,, and fpcore can be synthesize from an operator
+(define (register-operator-impl! op
+                                 name
+                                 args
+                                 orepr
+                                 #:fl [fl #f]
+                                 #:spec [spec #f]
+                                 #:fpcore [fpcore #f])
+  ;; Check if spec is given (if not, infer it from the operator which is required)
+  (define vars (map car args))
```

Comment:
Variables should be unique.

Reply:
Could be done here. I think it theoretically be done in the macro actually.

Reply:
Just gonna do it here